src/core/Akka/Actor/ActorRef.cs

```diff
@@ -121,43 +104,37 @@ public override IActorRefProvider Provider
         /// <param name="sender">TBD</param>
         protected override void TellInternal(object message, IActorRef sender)
         {
+            var handled = false;
 
-            if (message is ISystemMessage sysM) //we have special handling for system messages
-            {
-                SendSystemMessage(sysM);
-            }
-            else
+            switch (message)
             {
-                if (Interlocked.Exchange(ref status, COMPLETED) == INITIATED)
-                {
-                    if (message is T t)
-                    {
-                        _result.TrySetResult(t);
-                    }
-                    else if (message == null) //special case: https://github.com/akkadotnet/akka.net/issues/5204
-                    {
-                        _result.TrySetResult(default);
-                    }
-                    else if (message is Failure f)
-                    {
-                        _result.TrySetException(f.Exception ?? new TaskCanceledException("Task cancelled by actor via Failure message."));
-                    }
-                    else
-                    {
-                        _result.TrySetException(new ArgumentException(
-                            $"Received message of type [{message.GetType()}] - Ask expected message of type [{typeof(T)}]"));
-                    }
-                }
+                case ISystemMessage sysM:
+                    SendSystemMessage(sysM); //we have special handling for system messages
```

Comment:
I would throw an exception here - not supposed to send any system messages to temporary actors